app/interactors/recognize_user/authenticate_by_jwt.rb

```diff
@@ -0,0 +1,19 @@
+class RecognizeUser
+  class AuthenticateByJwt
+    include Interactor
+
+    delegate :token, to: :context
+
+    def call
+      user_attributes = JwtCodec.new.decode(token)
+      fail! unless user_attributes
+      context.user = User.new(user_attributes)
+    end
```

Comment:
```suggestion
    def call
      payload = JwtCodec.new.decode(token)
      context.user = User.find(payload["id"])
    end
```

Also, I think we can move these lines to the controller. It doesn't look too long.

Reply:
I don't want to, because I think when it's time to implement refresh-token-functionality I'll have to move it to interactors.